### ISSUE TITLE
Possible issue: packages corrupted after upload?

### ISSUE THREAD

**Issue by ecanault** [deploymenttheory/terraform-provider-jamfpro#485]
Hi,

I continue my testings and found that packages seems to be corrupted when uploaded with terraform:
- Installations fail
- Downloaded package from Jamf Pro console does not install locally
- Getting the package from a Cache action from a policy produces the same error

<img width="318" alt="image" src="https://github.com/user-attachments/assets/8ee69a8e-0572-4d7c-b2dc-dfe6b9573ba4">

I tried with 2 Jamf Pro console (11.11.1), each time using 2 different packages.
I tried also to upload the package from locally and remotely. 

Here is my sample code:

```
resource "jamfpro_package" "jamfpro_package_001" {
  package_name = "Setup.Manager.1.1-496.pkg"
  #   package_file_source = "https://github.com/Jamf-Concepts/Setup-Manager/releases/download/v1.1/Setup.Manager.1.1-496.pkg"
  package_file_source   = "./Setup.Manager.1.1-496.pkg"
  category_id           = "-1"
  info                  = "tf package deployment for demonstration"
  notes                 = "Uploaded by: terraform-provider-jamfpro plugin."
  priority              = 10
  reboot_required       = false
  fill_user_template    = false
  fill_existing_users   = false
  os_requirements       = ""
  swu                   = false
  self_heal_notify      = false
  os_install            = false
  serial_number         = ""
  suppress_updates      = false
  ignore_conflicts      = false
  suppress_from_dock    = false
  suppress_eula         = false
  suppress_registration = false
  manifest              = ""
  manifest_file_name    = ""
#  timeouts {
#    create = "90m"
#  }
```

Please could you have a look or tell me if I'm doing something wrong?

Thanks in advance,
Emmanuel

**Comment by ShocOne** [comment]
thanks, i have been able to replicate the issue. another release will be coming tomorrow with the fixes included

**Comment by ShocOne** [comment]
these fixes are now within v0.6.0. thank you for the feedback. We have added additional checks for package deployments comparing before and after the sha3_512 value of the file, to ensure broken uploads are identified early.

**Comment by ecanault** [comment]
You're welcome!
